common/lib/xmodule/xmodule/js/src/video/08_video_speed_control.js

```diff
@@ -77,31 +106,34 @@ function (Iterator) {
          * mousemove, etc.).
          */
         bindHandlers: function () {
-            var self = this;
-
             // Attach various events handlers to the speed menu button.
             this.el.on({
-                'mouseenter': this.mouseEnterHandler.bind(this),
-                'mouseleave': this.mouseLeaveHandler.bind(this),
-                'click': this.clickMenuHandler.bind(this),
-                'keydown': this.keyDownMenuHandler.bind(this)
+                'mouseenter': this.mouseEnterHandler,
+                'mouseleave': this.mouseLeaveHandler,
+                'click': this.clickMenuHandler,
+                'keydown': this.keyDownMenuHandler
             });
 
             // Attach click and keydown event handlers to the individual speed
             // entries.
             this.speedsContainer.on({
-                click: this.clickLinkHandler.bind(this),
-                keydown: this.keyDownLinkHandler.bind(this)
+                click: this.clickLinkHandler,
+                keydown: this.keyDownLinkHandler
             }, 'a.speed-link');
 
             this.state.el.on({
-                'speed:set': function (event, speed) {
-                    self.setSpeed(speed, true);
-                },
-                'speed:render': function (event, speeds, currentSpeed) {
-                    self.render(speeds, currentSpeed);
-                }
+                'speed:set': this.onSetSpeed,
+                'speed:render': this.onRenderSpeed
             });
+            this.state.el.on('destroy', this.destroy);
+        },
+
+        onSetSpeed: function (event, speed) {
+            this.setSpeed(speed, true);
```

Comment:
> Doesn't setSpeed take only one parameter now?

it's local for the module `setSpeed` method (it is not the same as `this.state.setSpeed`).